test/test-mongocrypt-ctx-encrypt.c

```diff
@@ -4856,6 +4855,944 @@ static void _test_fle2_collinfo_with_bad_str_encode_version(_mongocrypt_tester_t
     mongocrypt_destroy(crypt);
 }
 
+static void _test_lookup(_mongocrypt_tester_t *tester) {
+    // Test $lookup works.
+#define TF(suffix) TEST_FILE("./test/data/lookup/csfle/" suffix)
+    {
+        mongocrypt_t *crypt = _mongocrypt_tester_mongocrypt(TESTER_MONGOCRYPT_DEFAULT);
+        mongocrypt_ctx_t *ctx = mongocrypt_ctx_new(crypt);
+
+        ASSERT_OK(mongocrypt_ctx_encrypt_init(ctx, "db", -1, TF("cmd.json")), ctx);
+
+        expect_and_reply_to_ismaster(ctx);
+
+        ASSERT_STATE_EQUAL(mongocrypt_ctx_state(ctx), MONGOCRYPT_CTX_NEED_MONGO_COLLINFO);
+        {
+            expect_mongo_op(ctx, TEST_BSON(BSON_STR({"name" : {"$in" : [ "c1", "c2" ]}})));
+            // Feed both needed schemas.
+            ASSERT_OK(mongocrypt_ctx_mongo_feed(ctx, TF("collInfo-c1.json")), ctx);
+            ASSERT_OK(mongocrypt_ctx_mongo_feed(ctx, TF("collInfo-c2.json")), ctx);
+            ASSERT_OK(mongocrypt_ctx_mongo_done(ctx), ctx);
+        }
+
+        ASSERT_STATE_EQUAL(mongocrypt_ctx_state(ctx), MONGOCRYPT_CTX_NEED_MONGO_MARKINGS);
+        expect_mongo_op(ctx, TF("cmd-to-mongocryptd.json"));
+
+        mongocrypt_ctx_destroy(ctx);
+        mongocrypt_destroy(crypt);
+    }
+#undef TF
+
+    // Test $lookup errors if multiple-collection support is not opted-in.
+#define TF(suffix) TEST_FILE("./test/data/lookup/csfle/" suffix)
+    {
+        mongocrypt_t *crypt = _mongocrypt_tester_mongocrypt(TESTER_MONGOCRYPT_DEFAULT);
+        crypt->multiple_collinfo_enabled = false;
```

Comment:
```suggestion
```
This line does not appear to be necessary.

Reply:
Oop. I missed this comment on first pass. This line is needed since `_mongocrypt_tester_mongocrypt` enables the opt-in with a call to `mongocrypt_setopt_enable_multiple_collinfo`. This test expects an error due to the opt-in being disabled.

Reply:
It turns out my suggestion was due to a local test execution misconfiguration. Disregard, but also thank you for the explanation.